### PR TITLE
API tests

### PR DESCRIPTION
**Pull request by kgodey**
Related to #37, #38, #39

- Created separate test settings file to avoid modifying `settings.DATABASES` on the fly.
- Reordered `api` and `serializers` code to keep order of Schema, Table, Record the same across all files.
- Reconfigured fixtures a bit.
- Added tests for:
  - Schema list
  - Schema detail

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: mathesar/tests/settings.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+DATABASES = {
+    "default": {
+        "ENGINE": "django.db.backends.postgresql",
+        "NAME": "mathesar_django",
+        "USER": "mathesar",
+        "PASSWORD": "mathesar",
+        "HOST": "db",
+        "PORT": 5432,
+    },
+    "mathesar_db_test_database": {
```
Comment (kgodey) [inline mathesar/tests/settings.py:87]: I configured the test database here.

File: mathesar/serializers.py [hunk 2]
```diff
@@ -4,22 +4,17 @@
...
-
 class SchemaSerializer(serializers.HyperlinkedModelSerializer):
     class Meta:
         model = Schema
         fields = ['id', 'name', 'database', 'tables']
 
 
+class ColumnSerializer(serializers.Serializer):
+    name = serializers.CharField()
+    type = serializers.CharField()
```
Comment (mathemancer) [inline mathesar/serializers.py:15]: Will we also need to show other attributes (I'm thinking `fkey` should be slightly different from type, we might also want to note `primary_key` columns and whether they're `nullable`)

**Review by mathemancer — APPROVED** [review]
Okay, looks good!  I made a comment, but it's not blocking

File: mathesar/serializers.py (hunk 2 above)
Comment (kgodey) [reply mathesar/serializers.py:15]: Yeah, I think we'll want to show those later. I didn't make any changes to this code in this PR, I just moved it around.

**Pull request merged**